tests/components/airtouch5/test_cover.py

```diff
@@ -0,0 +1,114 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry, snapshot_platform
+
+COVER_ENTITY_ID = "cover.zone_1_damper"
+
+
+async def test_all_entities(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_airtouch5_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test all entities."""
+
+    with patch("homeassistant.components.airtouch5.PLATFORMS", [Platform.COVER]):
+        await setup_integration(hass, mock_config_entry)
+
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+async def test_cover_actions(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_airtouch5_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test the actions of the Airtouch5 covers."""
+
+    await setup_integration(hass, mock_config_entry)
+
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_CLOSE_COVER,
+        {ATTR_ENTITY_ID: COVER_ENTITY_ID},
+        blocking=True,
+    )
+    mock_airtouch5_client.send_packet.assert_called_once()
+    mock_airtouch5_client.reset_mock()
+
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_OPEN_COVER,
+        {ATTR_ENTITY_ID: COVER_ENTITY_ID},
+        blocking=True,
+    )
+    mock_airtouch5_client.send_packet.assert_called_once()
+    mock_airtouch5_client.reset_mock()
+
+    await hass.services.async_call(
+        COVER_DOMAIN,
+        SERVICE_SET_COVER_POSITION,
+        {ATTR_ENTITY_ID: COVER_ENTITY_ID, ATTR_POSITION: 50},
+        blocking=True,
+    )
+    mock_airtouch5_client.send_packet.assert_called_once()
+    mock_airtouch5_client.reset_mock()
+
+
+async def test_cover_callbacks(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    mock_airtouch5_client: AsyncMock,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test the callbacks of the Airtouch5 covers."""
+
+    await setup_integration(hass, mock_config_entry)
+
+    def to_zone_status_zone(open_percentage: int) -> ZoneStatusZone:
+        return ZoneStatusZone(
+            zone_power_state=ZonePowerState.ON,
+            zone_number=1,
+            control_method=ControlMethod.PERCENTAGE_CONTROL,
+            open_percentage=open_percentage,
+            set_point=None,
+            has_sensor=False,
+            temperature=None,
+            spill_active=False,
+            is_low_battery=False,
+        )
+
+    mock_airtouch5_client.zone_status_callbacks[0](data={1: to_zone_status_zone(70)})
+    state = hass.states.get(COVER_ENTITY_ID)
+    assert state
+    assert state.state == STATE_OPEN
+    assert state.attributes.get(ATTR_CURRENT_POSITION) == 70
```

Comment:
I'm after some help with this test. I don't think the call 'zone_status_callbacks[0]()' call is working at all. Maybe I didn't setup the mock correctly?

Reply:
If a callback is registered during runtime, you can get it from the mock call, checkout `knocki` for an example

Reply:
Cheers, I'll take a look.